client/src/views/client-area/dashboard/CustomOrderProcessingPage.vue

```diff
@@ -2,14 +2,143 @@
+              v-if="currentStatusStep > step.status"
+            />
+          </div>
+          <div class="process-step-text">
+            <ion-text class="process-step-text__title title-h3">
+              {{ $msTranslate(step.title) }}
+            </ion-text>
+            <ion-text
+              class="process-step-text__info body"
+              v-if="currentStatusStep === step.status"
+            >
+              {{ $msTranslate(step.description) }}
+            </ion-text>
+          </div>
+        </div>
+      </div>
+    </template>
+    <template v-else-if="querying">
+      <ms-spinner />
+    </template>
+    <template v-else-if="error || currentStatusStep === CustomOrderStatusStep.Unknown">
+      <ms-report-text :theme="MsReportTheme.Error">
+        {{ $msTranslate(error) }}
+      </ms-report-text>
+    </template>
   </div>
 </template>
 
 <script setup lang="ts">
-import { MsReportText, MsReportTheme } from 'megashark-lib';
+import { IonIcon, IonText } from '@ionic/vue';
+import { checkmarkCircle } from 'ionicons/icons';
+import { BmsAccessInstance, CustomOrderStatusResultData, DataType, BmsOrganization, CustomOrderStatus } from '@/services/bms';
+import { isDefaultOrganization } from '@/views/client-area/types';
+import { ref, onMounted } from 'vue';
+import { MsSpinner, MsReportText, MsReportTheme } from 'megashark-lib';
+import { getCustomOrderStatusTranslationKey } from '@/services/translation';
+
+const customOrderStatus = ref<CustomOrderStatusResultData>();
+const querying = ref(true);
+const error = ref('');
+
+const props = defineProps<{
+  organization: BmsOrganization;
+}>();
+
+enum CustomOrderStatusStep {
+  Unknown = 0,
+  NothingLinked = 1,
+  EstimateLinked = 2,
+  InvoiceToBePaid = 3,
+  InvoicePaid = 4,
+}
+
+const currentStatusStep = ref<CustomOrderStatusStep>(CustomOrderStatusStep.NothingLinked);
+
+const steps = [
+  {
+    status: CustomOrderStatusStep.NothingLinked,
+    title: getCustomOrderStatusTranslationKey(CustomOrderStatus.NothingLinked).title,
+    description: getCustomOrderStatusTranslationKey(CustomOrderStatus.NothingLinked).description,
+  },
+  {
+    status: CustomOrderStatusStep.EstimateLinked,
+    title: getCustomOrderStatusTranslationKey(CustomOrderStatus.EstimateLinked).title,
+    description: getCustomOrderStatusTranslationKey(CustomOrderStatus.EstimateLinked).description,
+  },
+  {
+    status: CustomOrderStatusStep.InvoiceToBePaid,
+    title: getCustomOrderStatusTranslationKey(CustomOrderStatus.InvoiceToBePaid).title,
+    description: getCustomOrderStatusTranslationKey(CustomOrderStatus.InvoiceToBePaid).description,
+  },
+  {
+    status: CustomOrderStatusStep.InvoicePaid,
+    title: getCustomOrderStatusTranslationKey(CustomOrderStatus.InvoicePaid).title,
+    description: getCustomOrderStatusTranslationKey(CustomOrderStatus.InvoicePaid).description,
+  },
+];
+
+onMounted(async () => {
+  if (isDefaultOrganization(props.organization)) {
```

Comment:
Set querying to true here, the component can be reused (and thus re-mounted) but not recreated.